packages/grid/_modules_/grid/components/menu/columnMenu/GridColumnHeaderMenu.tsx

```diff
@@ -25,7 +25,7 @@ export function GridColumnHeaderMenu({
   const currentColumn = apiRef.current.getColumn(field);
 
   const hideMenu = React.useCallback(
-    (event) => {
+    (event: React.SyntheticEvent<any, any>) => {
```

Comment:
This needs to be `React.SyntheticEvent<any>` because that function `hideMenu` is passed to the `GridMenu` and that on the other hand passes it to the `<ClickAwayListener onClickAway={hideMenu}>`. The `onClickAway` expects the event to be `React.SyntheticEvent<Document, MouseEvent>`